src/components/theme/View/View.jsx

```diff
@@ -203,8 +203,17 @@ class View extends Component {
    */
   render() {
     const { views } = config;
-    if (this.props.error && this.props.error.code === 301) {
-      const redirect = flattenToAppURL(this.props.error.url).split('?')[0];
+    const status = __SERVER__
+      ? this.props.error?.status
+      : this.props.error?.code;
+    // Checking to see if it's a 3XX HTTP status code. error.status only works on the server
+    if (status?.toString()?.[0] === '3') {
+      const redirectUrl = __SERVER__
+        ? this.props.error.response.headers.location
+        : this.props.error.url;
+      let redirect = flattenToAppURL(redirectUrl);
+      // We can hit situations where we end up being redirected to an api route. We don't want that so lets remove ++api++.
+      redirect = redirect.replace('/++api++', '');
```

Comment:
Strip querystring (see https://github.com/plone/volto/pull/4854#issuecomment-1689163422)

```suggestion
      redirect = redirect.replace('/++api++', '').replace(/\?.*/, '');
```